lib/matplotlib/__init__.py

```diff
@@ -890,6 +1057,7 @@ def _rc_params_in_file(fname, transform=lambda x: x, fail_on_error=False):
             raise
 
     config = RcParams()
+    config._parents()
```

Comment:
Please explain what this does and why we need it (at least in its docstring). The need to call a private method feels like a design flaw to me.

Reply:
Okay so this is a little ugly and I'm not very convinced either but the idea is - Since we're initializing `RcParams` with an empty dictionary before populating it with the default values, this leads to `RcParams` being initialized as basically `ChainMap({}, {})`. So, all the updates are then done in the first dictionary and the last dictionary just remains as an empty dictionary. So, to have the defaults in the last layer and have something like `ChainMap({}, defaults)` instead of `ChainMap({}, defaults, {})`, I remove the added layer and once all defaults have been set, I add a new layer.

Ideally, I would want to initialize it as `RcParams(defaults)` and not `RcParams()` and update then.